modules/compiler/vecz/source/include/vectorization_context.h

```diff
@@ -153,36 +153,59 @@ class VectorizationContext {
   /// @return The masked version of the function
   llvm::Function *getOrCreateMaskedFunction(llvm::CallInst *CI);
 
-  struct MaskedAtomicRMW {
+  /// @brief Represents either an atomicrmw or cmpxchg operation.
+  ///
+  /// Most fields are shared, with the exception of CmpXchgFailureOrdering and
+  /// IsWeak, which are only to be set for cmpxchg, and BinOp, which is only to
+  /// be set to a valid value for atomicrmw.
+  struct MaskedAtomic {
     llvm::Type *PointerTy;
     llvm::Type *ValTy;
+    /// @brief Must be set to BAD_BINOP for cmpxchg instructions
```

Comment:
It says above it should only be set to a valid value for atomicrmw?

Reply:
I'm taking `BAD_BINOP` to be an invalid binary operation. It you tried to create a `atomiccrmw` with `BAD_BINOP` you'd get an assert.